app/jobs/scheduled/create_recent_post_search_indexes.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Jobs
+  class CreateRecentPostSearchIndexes < ::Jobs::Scheduled
+    every 1.day
+
+    REGULAR_POST_SEARCH_DATA_INDEX_NAME = 'idx_recent_regular_post_search_data'
+
+    def execute(_)
+      create_recent_regular_post_search_index
+    end
+
+    private
+
+    def create_recent_regular_post_search_index
+      if !PostSearchData.where(private_message: false).offset(SiteSetting.search_enable_recent_regular_posts_offset_size - 1).limit(1).exists?
+        return
+      end
+
+      SiteSetting.search_recent_regular_posts_offset_post_id = offset_post_id(private_message: false)
+
+      DB.exec(<<~SQL)
+      DROP INDEX IF EXISTS temp_idx_recent_regular_post_search_data;
+      SQL
+
+      DB.exec(<<~SQL)
+      CREATE INDEX #{Rails.env.test? ? '' : 'CONCURRENTLY'} temp_idx_recent_regular_post_search_data
+      ON post_search_data USING GIN(search_data)
+      WHERE NOT private_message AND post_id >= #{SiteSetting.search_recent_regular_posts_offset_post_id}
+      SQL
+
+      DB.exec(<<~SQL)
+      #{Rails.env.test? ? '' : "BEGIN;"}
+      DROP INDEX IF EXISTS #{REGULAR_POST_SEARCH_DATA_INDEX_NAME};
+      ALTER INDEX temp_idx_recent_regular_post_search_data RENAME TO #{REGULAR_POST_SEARCH_DATA_INDEX_NAME};
+      #{Rails.env.test? ? '' : "COMMIT;"}
+      SQL
+    end
+
+    def offset_post_id(private_message:)
```

Comment:
This function is only ever called once with `false` as the value so why make it a parameter?

Reply:
agree... lest just call it `regular_offset_post_id`